.github/workflows/pull_request_checks.yml

```diff
@@ -0,0 +1,57 @@
+name: Pull Request Checks
+
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
```

Comment:
This line is why we need a separate file - this causes this job to be re-run whenever someone edits the PR title/description. If we had the debug builds in this file, they would run unnecessarily, costing resources.

Reply:
Can you add a comment explaining that. We will forget.